src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,400 @@
+  rw [← of_diff hA hB h, add_sub_cancel'],
+  apply_instance,
+end
+
+lemma of_Union_nonneg {M : Type*} [topological_space M]
+  [ordered_add_comm_monoid M] [order_closed_topology M]
+  {v : vector_measure α M} (hf₁ : ∀ i, measurable_set (f i))
+  (hf₂ : pairwise (disjoint on f)) (hf₃ : ∀ i, 0 ≤ v (f i)) :
+  0 ≤ v (⋃ i, f i) :=
+(v.of_disjoint_Union hf₁ hf₂).symm ▸ tsum_nonneg hf₃
+
+lemma of_Union_nonpos {M : Type*} [topological_space M]
+  [ordered_add_comm_monoid M] [order_closed_topology M]
+  {v : vector_measure α M} (hf₁ : ∀ i, measurable_set (f i))
+  (hf₂ : pairwise (disjoint on f)) (hf₃ : ∀ i, v (f i) ≤ 0) :
+  v (⋃ i, f i) ≤ 0 :=
+(v.of_disjoint_Union hf₁ hf₂).symm ▸ tsum_nonpos hf₃
+
+lemma of_nonneg_disjoint_union_eq_zero {s : signed_measure α} {A B : set α}
+  (h : disjoint A B) (hA₁ : measurable_set A) (hB₁ : measurable_set B)
+  (hA₂ : 0 ≤ s A) (hB₂ : 0 ≤ s B)
+  (hAB : s (A ∪ B) = 0) : s A = 0 :=
+begin
+  rw of_union h hA₁ hB₁ at hAB,
+  linarith,
+  apply_instance,
+end
+
+lemma of_nonpos_disjoint_union_eq_zero {s : signed_measure α} {A B : set α}
+  (h : disjoint A B) (hA₁ : measurable_set A) (hB₁ : measurable_set B)
+  (hA₂ : s A ≤ 0) (hB₂ : s B ≤ 0)
+  (hAB : s (A ∪ B) = 0) : s A = 0 :=
+begin
+  rw of_union h hA₁ hB₁ at hAB,
+  linarith,
+  apply_instance,
+end
+
+lemma summable_measure {μ : measure α} [hμ : finite_measure μ]
+  (hf₁ : ∀ (i : ℕ), measurable_set (f i))
+  (hf₂ : pairwise (disjoint on f)) :
+  summable (λ x, (μ (f x)).to_real) :=
+begin
+  apply ennreal.summable_to_real,
+  rw ← measure_theory.measure_Union hf₂ hf₁,
+  exact ne_of_lt (measure_lt_top _ _)
+end
+
+end vector_measure
+
+namespace measure
+
+/-- A finite measure coerced into a real function is a signed measure. -/
+def to_signed_measure (μ : measure α) [hμ : finite_measure μ] : signed_measure α :=
+{ measure_of := λ i : set α, if measurable_set i then (μ.measure_of i).to_real else 0,
+  empty := by simp [μ.empty],
+  not_measurable := λ _ hi, if_neg hi,
+  m_Union :=
+  begin
+    intros _ hf₁ hf₂,
+    rw [μ.m_Union hf₁ hf₂, ennreal.tsum_to_real_eq, if_pos (measurable_set.Union hf₁),
+        summable.has_sum_iff],
+    { congr, ext n, rw if_pos (hf₁ n) },
+    { refine @summable_of_nonneg_of_le _ (ennreal.to_real ∘ μ ∘ f) _ _ _ _,
+      { intro, split_ifs,
+        exacts [ennreal.to_real_nonneg, le_refl _] },
+      { intro, split_ifs,
+        exacts [le_refl _, ennreal.to_real_nonneg] },
+        exact vector_measure.summable_measure hf₁ hf₂ },
+    { intros a ha,
+      apply ne_of_lt hμ.measure_univ_lt_top,
+      rw [eq_top_iff, ← ha, outer_measure.measure_of_eq_coe, coe_to_outer_measure],
+      exact measure_mono (set.subset_univ _) }
+  end }
+
+lemma to_signed_measure_apply_measurable {μ : measure α} [finite_measure μ]
+  {i : set α} (hi : measurable_set i) :
+  μ.to_signed_measure i = (μ i).to_real :=
+if_pos hi
+
+/-- A measure is a vector measure over `ℝ≥0∞`. -/
+def to_vector_measure (μ : measure α) : vector_measure α ℝ≥0∞ :=
```

Comment:
the name feels strange, because you are not mapping it to a generic vector measure, but to one on `ennreal`. To emphasize this, maybe a more specific name like `to_ennreal_vector_measure` would make sense. (The name is long, but since I don't expect this will be used a lot this shouldn't be a problem).